src/ore/src/tracing.rs

```diff
@@ -218,6 +252,14 @@ where
         .with_ansi(!no_color && atty::is(atty::Stream::Stderr))
         .with_filter(config.stderr_log.filter);
 
+    let (stderr_log_layer, stderr_reloader) = reload::Layer::new(stderr_log_layer);
```

Comment:
I agree its unclear if this makes any measurable difference, but i'd still prefer we add a flag like `--dynamic-filter` to turn this on and off

the annoying thing will be that youll have turn the second half of this function into its own generic function, called with either the `fmt` layer or the `reload(fmt)` layer, depending which can be difficult, let me see if I can find an example

alternatively, you can just box the layer: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Layer.html#impl-Layer%3CS%3E-for-Box%3Cdyn%20Layer%3CS%3E%20+%20Send%20+%20Sync%3E-1

Reply:
Oh yeah, here are the generics: https://github.com/MaterializeInc/materialize/commit/29c2669311524a740e0a18a2710da9437807896f#diff-be4f91974540150a083d2e18d1029742f81be7fe81e8cdcc2a9430411fce98a3R55-R64

nightmare `Layered` stuff with `LookupSpan` and `SubscriberInitExt` but it works!

Reply:
I'd love to avoid making the configurability configurable if at all possible. Either it _is_ performant enough that we can leave it on all the time, or it isn't and we have to leave it off pretty much all the time... in which case what value is it providing if you have to restart all the pods anyway to add the `--dynamic-filter` argument?

Reply:
:seemsreasonble: